src/core/ext/filters/stateful_session/stateful_session_filter.cc

```diff
@@ -158,29 +164,44 @@ ArenaPromise<ServerMetadataHandle> StatefulSessionFilter::MakeCallPromise(
               this, cookie_config->name->c_str(),
               std::string(*cookie_value).c_str());
     }
-    // We have a valid cookie, so add the call attribute to be used by the
-    // xds_override_host LB policy.
-    service_config_call_data->SetCallAttribute(
-        GetContext<Arena>()->New<XdsOverrideHostAttribute>(*cookie_value));
+    std::pair<absl::string_view, absl::string_view> host_cluster =
+        absl::StrSplit(*cookie_value, absl::MaxSplits(';', 1));
+    if (!host_cluster.first.empty()) {
+      // We have a valid cookie, so add the call attribute to be used by the
```

Comment:
Here we need to check that the cluster specified in the cookie is actually present in the selected route.